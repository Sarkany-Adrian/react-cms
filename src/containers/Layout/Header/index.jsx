// @flow
import React, { type Node } from 'react';
// style
import './styles.scss';

type Props = {
  left: Node,
  center?: Node,
  right: Node
};

function Header({ left, center, right }: Props): React$Element<'div'> {
  return (
    <div className="app-header">
      <div className="h-100 row">
        <div className="app-header__left col-12 col-sm-4 col-md-4 col-lg-3 col-xl-3">
          {left}
        </div>
        <div
          className={`app-header__center col-12 col-sm-4 col-md-4 col-lg-6 col-xl-6 ${
            !center ? 'd-none d-sm-block' : ''
          }`}
        >
          {center}
        </div>
        <div className="app-header__right col-12 col-sm-4 col-md-4 col-lg-3 col-xl-3">
          {right}
        </div>
      </div>
    </div>
  );
}

Header.defaultProps = {
  center: ''
};

export default Header;
