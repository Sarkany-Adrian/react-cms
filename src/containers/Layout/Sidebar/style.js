// @flow
import styled from 'styled-components';

export const StyledSidebar = styled.aside`
  font-family: ${({ theme }: any) => theme.font.fontBody};
  height: 100%;
  z-index: 1;
  top: 0;
  left: 0;
  position: fixed;
  box-shadow: 0 4px 10px 0 rgba(0, 0, 0, 0.2), 0 4px 20px 0 rgba(0, 0, 0, 0.19);
  box-sizing: border-box;
  overflow: hidden;
`;

export const StyledSidebarHeader = styled.div`
  background-color: ${({ theme }: any) => theme.colors.primary};
  color: white;
  padding: 16px;
  text-align: center;
`;

export const StyledMenuItem = styled.li`
  display: block;
  padding: 16px 0;
  color: blue;
  text-decoration: none;

  a {
    display: block;
    padding: 16px 0;
    color: ${({ theme }: any) => theme.baseColors.lightBlue};
    text-decoration: none;
  }
`;
