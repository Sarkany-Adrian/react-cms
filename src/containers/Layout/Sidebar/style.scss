@import '~styles/resources.scss';

.app-sidebar {
  border: 1px solid $color-black;
}
