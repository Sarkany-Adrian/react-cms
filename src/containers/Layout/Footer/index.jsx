// @flow
import React, { Children, type Node } from 'react';
import { chunk } from 'helpers/array';
// style
import './style.scss';

type TProps = {
  children?: Node,
  rows?: number
};

function Footer({ children, rows }: TProps): React$Element<'div'> {
  const numberOfRows = Children.count(children) / Number(rows);
  let childArr = Children.toArray(children);
  childArr = chunk(childArr, numberOfRows);

  return (
    <div className="app-footer container">
      {childArr.map((row: Array<React$Element<*>>) => (
        <div key={row[0].key} className="app-footer__row row">
          {row.map((item: React$Element<*>) => (
            <div key={item.key} className="col-sm col-md">
              {item}
            </div>
          ))}
        </div>
      ))}
    </div>
  );
}

Footer.defaultProps = {
  rows: 1,
  children: []
};

export default Footer;
