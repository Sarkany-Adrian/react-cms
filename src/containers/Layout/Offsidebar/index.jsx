// @flow
import React, { PureComponent } from 'react';
import onClickOutside from 'react-onclickoutside';
import cx from 'classnames';

// components
import Button from 'atoms/Button';

// style
import './style.scss';

// fonts
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

// types
type Props = {
  isOpen: boolean,
  toggleOffsidebar: () => void,
  HOCtoggleOffsidebar: boolean => void
};

/**
 *
 */
export class Offsidebar extends PureComponent<Props> {
  /**
   * Used by on click outside wrapper
   */
  handleClickOutside = () => {
    const { HOCtoggleOffsidebar } = this.props;
    HOCtoggleOffsidebar(false);
  };

  /**
   *
   * @returns {*}
   */
  render() {
    const { isOpen, toggleOffsidebar } = this.props;

    return (
      <aside
        className={cx(
          'app-offsidebar',
          isOpen ? 'app-offsidebar--open' : 'app-offsidebar--closed'
        )}
      >
        <div className="app-offsidebar__header">
          <Button
            className="app-offsidebar__header__close"
            onClick={toggleOffsidebar}
          >
            <FontAwesomeIcon
              icon={faTimes}
              className="app-offsidebar__header__close__icon ignore-react-onclickoutside"
            />
          </Button>
          Stuff here
        </div>
        <div className="app-offsidebar__separator" />
      </aside>
    );
  }
}

export default onClickOutside(Offsidebar);
