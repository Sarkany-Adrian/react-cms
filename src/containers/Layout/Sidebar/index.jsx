// @flow
import React, { PureComponent } from 'react';
import onClickOutside from 'react-onclickoutside';
// theme
import { ThemeProvider } from 'styled-components';
import { SiteThemeContext } from 'contexts/theme';
// components
import SidebarBody from 'containers/Layout/Sidebar/Components/SidebarBody';
// style
import './style.scss';

type Props = {
  isOpen: boolean,
  toggleSidebar: boolean => void
};

export class Sidebar extends PureComponent<Props> {
  handleClickOutside = () => {
    const { toggleSidebar } = this.props;
    toggleSidebar(false);
  };

  render() {
    const { isOpen } = this.props;

    return (
      <SiteThemeContext.Consumer>
        {(props: any) => (
          <ThemeProvider theme={props.theme}>
            <SidebarBody isOpen={isOpen} onClick={this.handleClickOutside} />
          </ThemeProvider>
        )}
      </SiteThemeContext.Consumer>
    );
  }
}

export default onClickOutside(Sidebar);
