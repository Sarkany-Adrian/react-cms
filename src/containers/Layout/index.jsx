// @flow
import React, { Component } from 'react';
import type { ComponentType } from 'react';
import { Route, withRouter } from 'react-router-dom';
import cx from 'classnames';
// components
import Header from 'containers/Layout/Header';
import Footer from 'containers/Layout/Footer';
import ConnectedSidebar from 'containers/Layout/Sidebar';
import LeftSection from 'atoms/HeaderSections/LeftSection';
import RightSection from 'atoms/HeaderSections/RightSection';
// mobx
import { observer, inject } from 'mobx-react';
// style
import './style.scss';

type Props = {
  component: ComponentType<*>,
  path: string,
  exact: ?boolean,
  auth: Object,
  location: Object
};

type State = {
  isSidebarOpen: boolean
};

@inject(['auth'])
@observer
class Layout extends Component<Props, State> {
  state = {
    isSidebarOpen: false
  };

  toggleSidebar = (isOpen: boolean) => {
    this.setState({ isSidebarOpen: isOpen });
  };

  render() {
    const { component: Comp, auth, location, ...rest } = this.props;
    const { isSidebarOpen } = this.state;

    return (
      <div className="app-wrapper">
        <ConnectedSidebar
          isOpen={isSidebarOpen}
          toggleSidebar={this.toggleSidebar}
        />
        <div
          className={cx(
            'app-content',
            isSidebarOpen
              ? 'app-content--sidebar-open'
              : 'app-content--sidebar-closed'
          )}
        >
          <Header
            left={
              <LeftSection
                onClickMenu={this.toggleSidebar}
                onClickProfile={this.toggleSidebar}
              />
            }
            right={
              <RightSection
                onClickSettings={this.toggleSidebar}
                onClickProfile={this.toggleSidebar}
              />
            }
          />
          <div className="app-content">
            <Route {...rest} component={Comp} />
            <Footer />
          </div>
        </div>
      </div>
    );
  }
}

export default withRouter(Layout);
