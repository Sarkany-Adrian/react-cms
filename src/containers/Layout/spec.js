import React from 'react';
import { shallow } from 'enzyme';

import Layout from 'containers/Layout';

describe('Layout container', () => {
  it('should trigger the toggle sidebar method', () => {
    const wrapper = shallow(<Layout.wrappedComponent />);
    wrapper.instance().toggleSidebar(true);
    expect(wrapper.instance().state.isSidebarOpen).toEqual(true);
  });

  it('should be able to toggle the sidebar with onclickoutside method', () => {
    const wrapper = shallow(<Layout.wrappedComponent />);
    wrapper.instance().HOCtoggleSidebar(true);
    expect(wrapper.instance().state.isSidebarOpen).toBe(true);
  });

  it('should be able to toggle the offsidebar method', () => {
    const wrapper = shallow(<Layout.wrappedComponent />);
    wrapper.instance().toggleOffsidebar(true);
    expect(wrapper.instance().state.isOffsidebarOpen).toBe(true);
  });

  it('should be able to toggle the offsidebar with onclickoutside method', () => {
    const wrapper = shallow(<Layout.wrappedComponent />);
    wrapper.instance().HOCtoggleOffsidebar(true);
    expect(wrapper.instance().state.isOffsidebarOpen).toBe(true);
  });
});
