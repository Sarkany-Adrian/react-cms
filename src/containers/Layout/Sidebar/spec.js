import React from 'react';
import { shallow, mount } from 'enzyme';
import { Sidebar } from 'containers/Layout/Sidebar';
import {
  StyledSidebar,
  StyledSidebarHeader,
  StyledMenuItem
} from 'containers/Layout/Sidebar/style';

const theme = {
  font: {
    fontBody: 'test'
  },
  colors: {
    primary: 'red'
  },
  baseColors: {
    lightBlue: 'blue'
  }
};

describe('App Sidebar', () => {
  const mockedProps = {
    isOpen: true,
    HOCtoggleSidebar: jest.fn()
  };

  it('should call toggleSidebar', () => {
    const wrapper = shallow(<Sidebar {...mockedProps} />);
    wrapper.instance().handleClickOutside();
    expect(mockedProps.HOCtoggleSidebar).toHaveBeenCalledTimes(1);
  });
});

describe('Styled components', () => {
  it('Should render StyledSidebar', () => {
    const wrapper = mount(<StyledSidebar theme={theme} />);
    expect(wrapper.find('aside').exists()).toBe(true);
  });

  it('Should render StyledSidebarHeader', () => {
    const wrapper = mount(<StyledSidebarHeader theme={theme} />);
    expect(wrapper.find('div').exists()).toBe(true);
  });

  it('Should render StyledMenuItem', () => {
    const wrapper = mount(<StyledMenuItem theme={theme} />);
    expect(wrapper.find('li').exists()).toBe(true);
  });
});
