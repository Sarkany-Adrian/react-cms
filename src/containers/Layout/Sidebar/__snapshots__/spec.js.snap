// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Sidebar when the sidebar is closed, renders and matches snapshot 1`] = `
<aside
  className="app-sidebar app-sidebar--closed"
>
  <div
    className="app-sidebar__header"
  >
    Menu
    <Button
      className="app-sidebar__header__close"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="app-sidebar__header__close__icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              352,
              512,
              Array [],
              "f00d",
              "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
            ],
            "iconName": "times",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Button>
  </div>
  <div
    className="app-sidebar__user-block"
  >
    user block
  </div>
  <div
    className="app-sidebar__separator"
  />
  <ul
    className="app-sidebar__item-list"
  >
    <li
      className="app-sidebar__item-list__item"
      key="1"
    >
      <Link
        replace={false}
        to="/page1"
      >
        page1
      </Link>
    </li>
    <li
      className="app-sidebar__item-list__item"
      key="2"
    >
      <Link
        replace={false}
        to="/page2"
      >
        page2
      </Link>
    </li>
    <li
      className="app-sidebar__item-list__item"
      key="3"
    >
      <Link
        replace={false}
        to="/library"
      >
        Library
      </Link>
    </li>
  </ul>
</aside>
`;

exports[`App Sidebar when the sidebar is open, renders and matches snapshot 1`] = `
<aside
  className="app-sidebar app-sidebar--open"
>
  <div
    className="app-sidebar__header"
  >
    Menu
    <Button
      className="app-sidebar__header__close"
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className="app-sidebar__header__close__icon"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              352,
              512,
              Array [],
              "f00d",
              "M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z",
            ],
            "iconName": "times",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Button>
  </div>
  <div
    className="app-sidebar__user-block"
  >
    user block
  </div>
  <div
    className="app-sidebar__separator"
  />
  <ul
    className="app-sidebar__item-list"
  >
    <li
      className="app-sidebar__item-list__item"
      key="1"
    >
      <Link
        replace={false}
        to="/page1"
      >
        page1
      </Link>
    </li>
    <li
      className="app-sidebar__item-list__item"
      key="2"
    >
      <Link
        replace={false}
        to="/page2"
      >
        page2
      </Link>
    </li>
    <li
      className="app-sidebar__item-list__item"
      key="3"
    >
      <Link
        replace={false}
        to="/library"
      >
        Library
      </Link>
    </li>
  </ul>
</aside>
`;
